ksa_compliance/ksa_compliance/doctype/zatca_phase_1_business_settings/zatca_phase_1_business_settings.js

```diff
@@ -18,4 +17,23 @@ frappe.ui.form.on("ZATCA Phase 1 Business Settings", {
             }
         });
     },
+    address(frm){
```

Comment:
This should be `company`, not `address`. When a company is selected, we update the query to filter addresses based on it